### PR TITLE
Hook up gossip as slf4j backend and implement customized color renderer

### PR DESCRIPTION
**Pull request by jdillon**
Replaces slf4j-simple with gossip-slf4j ( https://github.com/jdillon/gossip ) and adds support for colorized output (via https://github.com/fusesource/jansi ).

maven-color

To enable colorized output:

```
MAVEN_OPTS="-Dmaven.logging=color" bin/mvn
```

or set in ~/.mavenrc

Presently has to be passed in via MAVEN_OPTS as there is no context to access the -D flags early enough to configure the backend w/o more changes to the Slf4jConfiguration api.

Default is presently plain, so users need to elect to use color explicitly as described above.

Colorization is a subset of the color handling available in mvnsh, as this can only colorize log events, so anything going directly to output streams (like bits from surefire, etc) can not be colorized by this mechanism.

Left the assembly conf/logging/* bits alone, as its not clear if other configurations are expecting this directory to exist or not, but likely should be cleaned up.

### CODE REVIEW

**Comment by jdillon** [comment]
Pending a groupId update to get a release of gossip which has a few minor bug fixes to make the core-it-suite happy.

File: pom.xml [hunk 1]
```diff
@@ -282,6 +282,16 @@
         <version>1.1.7</version>
         <optional>true</optional>
       </dependency>
+      <dependency>
+        <groupId>org.sonatype.gossip</groupId>
+        <artifactId>gossip-slf4j</artifactId>
+        <version>1.9-SNAPSHOT</version>
```
Comment (michael-o) [inline pom.xml]: This should resolve to a release version.

Comment (jdillon) [inline pom.xml]: Yup, working on that; marked the PR as WIP until I can get this sorted out

**Comment by michael-o** [comment]
What happens if the opts are set but `-l` or `>` is used? Do you detect that `System.out` is not a terminal?

**Comment by jdillon** [comment]
@michael-o w/o making more of the embedder aware of potential colorization its not easy to do.  I will push an example which should make > and -l work but it does involve adding a system property when the -l option is processed to strip out ansi.

The slf4j logging configuration presently does not have any context of the cli options, so its hard to make it fully aware of the configuration w/o such hacks, or potentially augmenting the api to configure.

**Comment by michael-o** [comment]
I know, that is why I have asked for. You may check out the link I have provided. This might help. Probably you should mention this limitation in your docs.

**Comment by jdillon** [comment]
@michael-o the jansi library has some limited support to detect tty when wrapping and potentially strip sequenced.  I've pushed a change which makes the gossip listener aware of ansi by default and will escape by default when using > and in the case of -l I had to add a minor hack to force it to strip via system property.

**Comment by michael-o** [comment]
It seems there is some magic: https://github.com/fusesource/jansi-native/blob/master/src/main/java/org/fusesource/jansi/internal/CLibrary.java#L59

**Comment by jdillon** [comment]
@michael-o yup, that is what is used internally by AnsiConsole.wrapOutputStream() to _help_ detect if it should strip sequences or not, but its not a proper public api to consume directly atm.

**Comment by michael-o** [comment]
Yes, leave it to JAnsi to decide. Just noticed that they do not provide a native package for FreeBSD. Will reach out to them next week.

**Comment by jdillon** [comment]
Updated gossip to 2.0.0 which is now in com.planet57 namespace and I can make a release of it shortly.

**Comment by jdillon** [comment]
Updated gossip to 2.0.0 release which is now in central.

**Comment by jvanzyl** [comment]
How are the ITs looking now? I know there were a few issues with the output not matching slf4j-simple that you were trying to sort out.

**Comment by jdillon** [comment]
@jvanzyl I believe the output issues were sorted, but was still seeing some odd failures with what looked like resolution mismatches, downloading a version of one artifact but ending up with a slightly different version.

**Comment by hboutemy** [comment]
I just committed the code, slightly modified since Maven general color support was added in MNG-3507 (after extracting some code from this provider)
I tested with core ITs and fixed one little issue in error rendering: there was one missing ":" after "Caused by": nothing complex, but this caused failures in ITs that were checking stacktrace content :)

Thank you Jason D: now we'll see if everybody is ok to use the provider as default...

File: maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/gossip/ColorRenderer.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                buff.append( ansi().a( INTENSITY_BOLD ).a( "at" ).reset().a( " " )
+                        .a( e.getClassName() ).a( "." ).a( e.getMethodName() ) );
+                buff.append( ansi().a( " (" ).a( INTENSITY_BOLD ).a( getLocation( e ) ).reset().a( ")" ) );
+                renderNewLine( buff );
+            }
+
+            cause = cause.getCause();
+            if ( cause != null )
+            {
+                buff.append( ansi().a( INTENSITY_BOLD ).a( "Caused by" ).reset().a( " " )
```
Comment (hboutemy) [inline maven-embedder/src/main/java/org/apache/maven/cli/logging/impl/gossip/ColorRenderer.java]: should be a( "Caused by" ).reset().a( ": " )
fixed while merging in Maven

**Comment by jira-importer** [comment]
Resolve #5408

**Pull request closed without merging**